### DISCUSSION TITLE
Problem using more than 8 MIDI_CC::General_Purpose_Controller on Leonardo

### DISCUSSION THREAD

**Discussion by xoanxil** [tttapa/Control-Surface#822] [General]
Hi,

I am trying to use a Leonardo board to create < small controller with 11 potenciometers and eigth switches. Now I am trying the pots but when I create more than 9 the code doesn´t recognize the following (General_Purpose_Controller appears black and not green) and I get error uploading.

This is the console error:

Varios_potenciometros_Arduino_Controller:51:17: error: 'General_Purpose_Controller_9' is not a member of 'CS::MIDI_CC'
   {A8, MIDI_CC::General_Purpose_Controller_9},
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xoanxil/Desktop/Varios_potenciometros_Arduino_Controller/Varios_potenciometros_Arduino_Controller.ino:51:17: note: suggested alternative: 'General_Purpose_Controller_8'
   {A8, MIDI_CC::General_Purpose_Controller_9},
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 General_Purpose_Controller_8
Varios_potenciometros_Arduino_Controller:52:17: error: 'General_Purpose_Controller_10' is not a member of 'CS::MIDI_CC'
   {A9, MIDI_CC::General_Purpose_Controller_10},
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xoanxil/Desktop/Varios_potenciometros_Arduino_Controller/Varios_potenciometros_Arduino_Controller.ino:52:17: note: suggested alternative: 'General_Purpose_Controller_1'
   {A9, MIDI_CC::General_Purpose_Controller_10},
                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                 General_Purpose_Controller_1
Varios_potenciometros_Arduino_Controller:53:18: error: 'General_Purpose_Controller_11' is not a member of 'CS::MIDI_CC'
   {A10, MIDI_CC::General_Purpose_Controller_11},
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/xoanxil/Desktop/Varios_potenciometros_Arduino_Controller/Varios_potenciometros_Arduino_Controller.ino:53:18: note: suggested alternative: 'General_Purpose_Controller_1'
   {A10, MIDI_CC::General_Purpose_Controller_11},
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                  General_Purpose_Controller_1
Varios_potenciometros_Arduino_Controller:54:1: error: could not convert '{A8, <expression error>}' from '<brace-enclosed initializer list>' to 'CS::CCPotentiometer'
 };
 ^
Varios_potenciometros_Arduino_Controller:54:1: error: could not convert '{A9, <expression error>}' from '<brace-enclosed initializer list>' to 'CS::CCPotentiometer'
Varios_potenciometros_Arduino_Controller:54:1: error: could not convert '{A10, <expression error>}' from '<brace-enclosed initializer list>' to 'CS::CCPotentiometer'
Usando librería Control-Surface-main con versión 2.0.0 en la carpeta: /Users/xoanxil/Documents/Arduino/libraries/Control-Surface-main 
Usando librería SPI con versión 1.0 en la carpeta: /Applications/Arduino.app/Contents/Java/hardware/arduino/avr/libraries/SPI 
Usando librería MIDIUSB con versión 1.0.5 en la carpeta: /Users/xoanxil/Documents/Arduino/libraries/MIDIUSB 
exit status 1
'General_Purpose_Controller_9' is not a member of 'CS::MIDI_CC'

#include <Control_Surface.h> // Include the Control Surface library

// Instantiate a MIDI over USB interface.
USBMIDI_Interface midi;

This is the code I am using:

CCPotentiometer knobsTop[] {
  {A0, MIDI_CC::General_Purpose_Controller_1},
  {A1, MIDI_CC::General_Purpose_Controller_2},
  {A2, MIDI_CC::General_Purpose_Controller_3},
  {A3, MIDI_CC::General_Purpose_Controller_4},
  {A4, MIDI_CC::General_Purpose_Controller_5},
  {A5, MIDI_CC::General_Purpose_Controller_6},
  {A6, MIDI_CC::General_Purpose_Controller_7},
  {A7, MIDI_CC::General_Purpose_Controller_8},
  {A8, MIDI_CC::General_Purpose_Controller_9},
  {A9, MIDI_CC::General_Purpose_Controller_10},
  {A10, MIDI_CC::General_Purpose_Controller_11}
};

void setup() {
  Control_Surface.begin(); // Initialize Control Surface
}

void loop() {
  Control_Surface.loop(); // Update the Control Surface
}

**Comment by tttapa** [comment]
`General_Purpose_Controller_9` does not exist. The MIDI specification only defines 8 general purpose controllers, see the specs on <https://midi.org>, or the list on https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/dbe/namespaceMIDI__CC.html.

You can use a different controller from the list, or use a number between 0 and 119.

**Reply by xoanxil** [reply]
Great. Good to know.  

Thank you very much for the information. It looks work propertly now.

Best
